src/Compilers/CSharp/Portable/Compilation/MemberSemanticModel.cs

```diff
@@ -1054,7 +1054,13 @@ private ImmutableArray<BoundNode> GuardedGetBoundNodesFromMap(CSharpSyntaxNode n
         {
             Debug.Assert(_nodeMapLock.IsWriteLockHeld || _nodeMapLock.IsReadLockHeld);
             ImmutableArray<BoundNode> result;
-            return _guardedNodeMap.TryGetValue(node, out result) ? result : default(ImmutableArray<BoundNode>);
+            return _guardedNodeMap.TryGetValue(AsNodeMapKey(node), out result) ? result : default(ImmutableArray<BoundNode>);
```

Comment:
I think this is not the right place to change the lookup key. This method should return bound nodes really associated with the given syntax node. I think the adjustments should be made in a different place. Either we should change binding to create a special bound node associated with EqualsValue syntax, or we should change initializer semantic model to never request a node for EqualsValue syntax.